src/components/LayerTree/LayerTree.js

```diff
@@ -299,6 +299,7 @@ class LayerTree extends Component {
 
   renderInput(layer, inputProps) {
     const { titles, isItemHidden } = this.props;
+    const { renderInputComponent, ...otherProps } = inputProps;
```

Comment:
use its own renderCheckbox prop like we have for the item content and the label.